### DISCUSSION TITLE
What service would you like us to use for YouTube redirects?

### DISCUSSION THREAD

**Discussion by nopeitsnothing** [Anon-Planet/thgtoa#308] [Polls]
# Invidious is scheduled to terminate services soon
This affects our ability to safely redirect you to more privacy-friendly things other than YouTube.

We need to figure out a more privacy friendly way to link to YouTube. For that, we have two paths: 

1. we could simply remove all Invidious redirects, and then ask readers to use LibRedirect on their own browser, thus leaving that risk to the reader themselves and 
3. we could switch all of the redirects to an alternative YouTube frontend, which may also break in the near furture, via API changes or code maintenance. 

I think we should go for option 1, placing the choice in others' hands, rather than changing hundreds of lines of text. It's mostly a discussion of what service and/or instances we should use, but all other services are less maintained than Invidous ever was. I'm just worried that in changing the links, they will need changing again in a few months or a year. Please discuss what service you would like to see us use, if any at all, or if you wouldn't mind using LibRedirect to put less stress on the devs to keep them updated over time.

Suppose if I had to choose I would go with [Piped](https://alternativeto.net/software/piped/about/) and the instance: https://piped-material.১.net/ (yes, it's a real link). I like FreeTube as well.

**Give any additional relevant context.**

### Links

We currently have 166 alternative links in superscript next to YT links. - [via my comment on Matrix](https://matrix.to/#/!gFXvNUxUwZAAFabSKX:matrix.anonymousplanet.org/$rxM2SpxaujZWHzMWQOM5IodPRnNNXxtdhoqwRfGWdqs?via=matrix.anonymousplanet.org&via=matrix.org&via=t2bot.io)

- https://torrentfreak.com/youtube-orders-invidious-privacy-software-to-shut-down-in-7-days-230609/
- https://alternativeto.net/news/2023/6/youtube-legal-team-asked-invidious-developers-to-take-down-the-service-within-7-days/
- https://torrentfreak.com/youtube-orders-invidious-privacy-software-to-shut-down-in-7-days-230609/

**Comment by sycam0r-e** [comment]
Finally came around to voting on this. I strongly feel that users should be educated once at the beginning of the guide about LibRedirect and then be encouraged to use it. Then, remove all redirects in the guide. The point that this destresses the devs and takes away this huge burden of possibly repeateadly replacing redirect links over time cannot be overemphasised. Having redirects in the guide is a nice thing sure but its benefit is tiny compared to the amount of work devs are going to have with replacing them, especially since LibRedirect exists and seems well maintained. I think users who read the guide are part of a population that will have no problem whatsoever with understanding and adopting the usage of LibRedirect.

**Comment by sycam0r-e** [comment]
This could also apply to other services, e.g. LibRedirect can also redirect Wikipedia links. I would minimise unnecessary dev work as much as possible. Changing any kinds of redirect links frequently seems like a waste of time of highly knowledgable and skilled THGOA devs that could be used for other more important and complicated things. Replace all redirect links with their original link and let LibRedirect do the work.
